### PR TITLE
updated readme file and added detailed description and examples

### PR DESCRIPTION
**Pull request by shahzadhaider1**
# Title: Documentation

## Description
This PR introduces a comprehensive update to the README file, making Utils more accessible and easier to use for developers. 

### Key improvements include:
- **Detailed Description**: Clear and concise explanations of each package within the library.
- **Usage Instructions**: Step-by-step guidance on installing and integrating Utils.
- **Examples for Core Utilities**: Real-world examples demonstrating how to use major utilities like boolean, ctxutils, maps, slice, strings, and structs.
- This update aims to provide developers with a robust starting point, showcasing the versatility of the Utils library and simplifying integration into their Go projects.

## Summary by CodeRabbit

- **Documentation**
	- Restructured and expanded the `readME.md` for improved clarity and usability.
	- Added new sections: "Overview," "Prerequisites," "Installation," "Key Features," "Usage Guide," and "Future Enhancements."
	- Included comprehensive example code snippets in the "Usage Guide" for practical application of library features.
	- Introduced a "Contributions" section to encourage community involvement.
	- Updated `CONTRIBUTING.md` with clearer instructions for forking and cloning the repository.

### CODE REVIEW

**Comment by shahzadhaider1** [comment]
@coderabbitai review

**Comment by shahzadhaider1** [comment]
@coderabbitai review

File: readME.md [hunk 1]
```diff
@@ -1,2 +1,183 @@
...
+```
+
+## Future Enhancements
+- Additional utilities for common string, slice, and map operations.
+- Customizable context key types and extended error handling.
+- Support for advanced encryption methods.
+
+
+# Contributions
+Contributions to this project are welcome! If you would like to contribute, please feel free to:
```
[the remark this answers is not in the document]
Comment (shahzadhaider1) [reply readME.md:193]: that looks incorrect, we can ignore this one

File: readME.md [hunk 2]
```diff
@@ -3,0 +1,183 @@
...
+To use utils in your project, add it as a module dependency:
+
+#### Clone the Repository
+```
+go get github.com/kashifkhan0771/utils
+```
+Alternatively, include it directly in your go.mod file:
+```
+require github.com/kashifkhan0771/utils latest
+```
```
[the remark this answers is not in the document]
Comment (shahzadhaider1) [reply readME.md]: resolved

[the remark this answers is not in the document]
Comment (shahzadhaider1) [reply readME.md]: addressed

**Comment by shahzadhaider1** [comment]
@coderabbitai review

**Review by kashifkhan0771 — APPROVED** [review]
LGTM! Thanks a lot for this @shahzadhaider1

**Review by kashifkhan0771 — APPROVED** [review]

**Pull request merged**